### DISCUSSION TITLE
BGP configuration in VRF topologies without bgp module

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1559] [General]
This Sunday, I was testing out topology [```integration/vrf/33-vrf-common-ospf```](https://github.com/ipspace/netlab/blob/dev/tests/integration/vrf/33-vrf-common-ospf.yml) while implementing Cumulus NVUE OSPFv2 support in VRFs

On Cumulus (and FRR in general, similar on EOS), this yields the following:
![image](https://github.com/user-attachments/assets/175ab151-d60e-441f-bb89-b28c33a3e47b)

The validation passes, but I'm seeing all sorts of BGP and IP-VPN related configs that would seem to require the bgp module - which isn't used in this topology. If not, it requires the ```vrf.bgp``` feature which isn't implemented for Cumulus NVUE - but there is no check/warning

Perhaps https://github.com/ipspace/netlab/blob/a1e1076bb4459eb85c75fe23bc13e2a7889a8fb9/netsim/modules/vrf.py#L280 should check for ```vpn.bgp``` support?

Thoughts?

**Comment by jbemmel** [comment]
See https://github.com/ipspace/netlab/pull/1560

**Comment by ipspace** [comment]
When you figure out how to do inter-VRF route leaking in FRR (and a bunch of other platforms) without going through BGP, please submit a PR (not promising I will merge it though, the current stuff works ;)

As for checking for yet another feature flag, there aren't that many platforms that cannot do inter-VRF route leaking, so that belongs into quirks.
